### PR TITLE
Support Type (tpe) header field

### PR DESCRIPTION
**Pull request by shaxbee**

### CODE REVIEW

**Comment by shaxbee** [comment]
Fixes #126

**Comment by shaxbee** [comment]
Travis seems to have some problem with task set-up, I've run tests locally:
```
> go test ./...
ok      gopkg.in/square/go-jose.v2      18.635s
ok      gopkg.in/square/go-jose.v2/cipher       0.105s
?       gopkg.in/square/go-jose.v2/jose-util    [no test files]
ok      gopkg.in/square/go-jose.v2/json 1.555s
ok      gopkg.in/square/go-jose.v2/jwt  0.158s
```

File: shared.go [hunk 1]
```diff
@@ -131,6 +131,7 @@ type rawHeader struct {
 	Jwk   *JSONWebKey          `json:"jwk,omitempty"`
 	Kid   string               `json:"kid,omitempty"`
 	Nonce string               `json:"nonce,omitempty"`
+	Tpe   string               `json:"tpe,omitempty"`
```
Comment (csstaub) [inline shared.go]: Shouldn't this be `typ` instead of `tpe`? 

See: https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-32#section-5.1

**Comment by csstaub** [comment]
FYI, I fixed the build failure on the master/v1/v2 branches, so if you rebase it should work again.

File: shared.go (hunk 1 above)
Comment (shaxbee) [inline shared.go]: Fixed.

**Comment by csstaub** [comment]
Looks good! Thank you @shaxbee

**Pull request merged**